src/usr/local/www/system_advanced_notifications.php

```diff
@@ -123,7 +143,12 @@
 			}
 		}
 
-		$config['notifications']['smtp']['authentication_mechanism'] = $_POST['smtpauthmech'];
+		if (!in_array($_POST['smtpauthmech'], $smtp_authentication_mechanisms)) {
```

Comment:
Seems like `$smtp_authentication_mechanisms` is missing a global declaration near the top of this file (and also missing in `/etc/inc/notices.inc` before it's defined).

Also this test should be using `array_key_exists()` rather than `in_array()`. It may work at the moment but only by coincidence since currently the key and value pairs in that array are identical.

Reply:
done, fixed

Reply:
global declaration outside a function makes no sense.  All variables are available on that scope

Reply:
It's defined in another file, not here, and isn't declared global before it's defined as global in either place. We've seen situations before where that has caused problems and the variable ends up empty in certain rare consequences.